### ISSUE TITLE
Make sample optional in the samplesheet

### ISSUE THREAD

**Issue by saulpierotti-ebi** [nf-core/sarek#1152]
### Description of feature

I often work with non-human germline data and in sarek it is required to specify a patient ID and a sample ID in the samplesheet. The IDs that are then written to the cram and VCF files are in the form patient_sample. In my case the concept of patient does not really apply and I just set patient and sample to the same value. However, this results in having IDs in the vcf and cram in the form ID_ID, while I would want them to be just ID. I would make patient optional when running germline variant calling.

**Comment by FriederikeHanssen** [comment]
or we could rename patient to entity?

**Comment by asp8200** [comment]
> or we could rename patient to entity?

Yeah, "entity" might be better. I was a bit puzzled by the choice of the column name "patient", since the sample does have to come from a patient. (Although I guess it is suitable for most users.)

**Comment by saulpierotti-ebi** [comment]
Renaming patient to entity I think makes sense but would not solve the sample naming issue in the outputs right? If entity is not optional the samples in the output will still be named entity_sample even for germline calling where this does not really make sense.

**Comment by pmoris** [comment]
I'm in the middle of exploring `sarek` for our (parasite) population genetics projects and I had exactly the same remark.

I'd be in favor of both options. I can image scenarios where multiple _samples_ can come from the same _subject_ (whether this be a patient, collection site, etc.), as well as other settings where _sample_ is a unique and top-level identifier that does not need to be nested further. 

The former would make the pipeline more general for analyses that do not involve patient data, but do have another grouping factor. I'd argue that _subject_ is more generally applicable than entity for this case, but I guess it could depend on your field.

The latter would fix the double naming issue for situations where _sample_ is the sole identifier and is guaranteed to be unique. It would also make the sarek pipeline more similar to the [rnaseq one](https://nf-co.re/rnaseq/3.12.0/docs/usage#samplesheet-input) (which only relies on `sample,fastq_1,fastq_2,strandedness`).

Another example where patient ID does not always make sense, is when (re)-analyzing reads from SRA. The [nf-core/fetchngs pipeline outputs a `samplesheet.csv`](https://nf-co.re/fetchngs/1.10.0/docs/usage#samplesheet-format), but it is currently only optimized for use with a few other nf-core pipelines (like nf-core/rnaseq). It seems to use a _sample_ column to store the SRA experiment and a `run` column to store the _SRA run_, although in typing this I wonder why the _BioSample_ identifier isn't being used as a sample column instead.

**Comment by FriederikeHanssen** [comment]
Thanks for the great input @pmoris 🙏 On a side note: making fetchngs directly output a suitable samplesheet is pretty high on our list of features we are working on in the next few months.

**Comment by Shaun-Regenbaum** [comment]
Did this ever end up going anywhere in regards to fetchngs?

**Comment by FriederikeHanssen** [comment]
sorry, no time yet. We#Ll look into it in the new year

**Comment by Shaun-Regenbaum** [comment]
I can work on it if you want. I am getting into nf-core workflows and contributing.

**Comment by FriederikeHanssen** [comment]
Yes sounds great. We have a dev channel called #sarek_dev on slack as well if you need help with anything :)
